The test test_partition_filters_specialstrings is failing on Windows

The test `fastparquet/test/test_partition_filters_specialstrings.py` is failing when the data is partitioned by `symbols` column and it's saved into a hive format. The column contains value `VIX*` which includes the special character `*` which is not allowed on Windows directory names. I'm not quite sure how to fix this one. One option would be to encode the character on Windows but it feels a little hacky unless it's done on all platforms.

Here is the output from the test for reference:
```
fastparquet/test/test_partition_filters_specialstrings.py::test_frame_write_read_verify[input_symbols3-504-hive-2-partitions3-filters3] FAILED                                                                                        [ 77%]
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

tempdir = 'C:\\Users\\Janne\\AppData\\Local\\Temp\\tmpf5ztngs4', input_symbols = ['VIX*', 'SPY', 'VIX'], input_days = 504, file_scheme = 'hive', input_columns = 2, partitions = ['symbol', 'year'], filters = [('symbol', '==', 'SPY')]

    @pytest.mark.parametrize('input_symbols,input_days,file_scheme,input_columns,'
                             'partitions,filters',
                             [
                                 (['NOW', 'SPY', 'VIX'], 2 * 252, 'hive', 2,
                                  ['symbol', 'year'], [('symbol', '==', 'SPY')]),
                                 (['now', 'SPY', 'VIX'], 2 * 252, 'hive', 2,
                                  ['symbol', 'year'], [('symbol', '==', 'SPY')]),
                                 (['TODAY', 'SPY', 'VIX'], 2 * 252, 'hive', 2,
                                  ['symbol', 'year'], [('symbol', '==', 'SPY')]),
                                 (['VIX*', 'SPY', 'VIX'], 2 * 252, 'hive', 2,
                                  ['symbol', 'year'], [('symbol', '==', 'SPY')]),
                                 (['QQQ*', 'SPY', 'VIX'], 2 * 252, 'hive', 2,
                                  ['symbol', 'year'], [('symbol', '==', 'SPY')]),
                                 (['QQQ!', 'SPY', 'VIX'], 2 * 252, 'hive', 2,
                                  ['symbol', 'year'], [('symbol', '==', 'SPY')]),
                                 (['Q%QQ', 'SPY', 'VIX'], 2 * 252, 'hive', 2,
                                  ['symbol', 'year'], [('symbol', '==', 'SPY')]),
                                 (['NOW', 'SPY', 'VIX'], 10, 'hive', 2,
                                  ['symbol', 'dtTrade'], [('symbol', '==', 'SPY')]),
                                 (['NOW', 'SPY', 'VIX'], 10, 'hive', 2,
                                  ['symbol', 'dtTrade'],
                                  [('dtTrade', '==',
                                    '2005-01-02T00:00:00.000000000')]),
                                 (['NOW', 'SPY', 'VIX'], 10, 'hive', 2,
                                  ['symbol', 'dtTrade'],
                                  [('dtTrade', '==',
                                    Timestamp('2005-01-01 00:00:00'))]),
                             ]
                             )
    def test_frame_write_read_verify(tempdir, input_symbols, input_days,
                                     file_scheme,
                                     input_columns, partitions, filters):
        # Generate Temp Director for parquet Files
        fdir = str(tempdir)
        fname = os.path.join(fdir, 'test')

        # Generate Test Input Frame
        input_df = frame_symbol_dtTrade_type_strike(days=input_days,
                                                    symbols=input_symbols,
                                                    numbercolumns=input_columns)
        input_df.reset_index(inplace=True)
        write(fname, input_df, partition_on=partitions, file_scheme=file_scheme,
>             compression='SNAPPY')

fastparquet\test\test_partition_filters_specialstrings.py:76:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
fastparquet\writer.py:877: in write
    with_field=file_scheme == 'hive'
fastparquet\writer.py:941: in partition_on_columns
    mkdirs(join_path(root_path, path))
fastparquet\util.py:34: in default_mkdirs
    os.makedirs(f, exist_ok=True)
c:\python\anaconda3\lib\os.py:210: in makedirs
    makedirs(head, mode, exist_ok)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'C:/Users/Janne/AppData/Local/Temp/tmpf5ztngs4/test/symbol=VIX*', mode = 511, exist_ok = True

    def makedirs(name, mode=0o777, exist_ok=False):
        """makedirs(name [, mode=0o777][, exist_ok=False])

        Super-mkdir; create a leaf directory and all intermediate ones.  Works like
        mkdir, except that any intermediate path segment (not just the rightmost)
        will be created if it does not exist. If the target directory already
        exists, raise an OSError if exist_ok is False. Otherwise no exception is
        raised.  This is recursive.

        """
        head, tail = path.split(name)
        if not tail:
            head, tail = path.split(head)
        if head and tail and not path.exists(head):
            try:
                makedirs(head, mode, exist_ok)
            except FileExistsError:
                # Defeats race condition when another thread created the path
                pass
            cdir = curdir
            if isinstance(tail, bytes):
                cdir = bytes(curdir, 'ASCII')
            if tail == cdir:           # xxx/newdir/. exists if xxx/newdir exists
                return
        try:
>           mkdir(name, mode)
E           OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:/Users/Janne/AppData/Local/Temp/tmpf5ztngs4/test/symbol=VIX*'

c:\python\anaconda3\lib\os.py:220: OSError```

--- a/fastparquet/test/test_partition_filters_specialstrings.py
+++ b/fastparquet/test/test_partition_filters_specialstrings.py
@@ -66,6 +66,9 @@ def frame_symbol_dtTrade_type_strike(days=1 * 252,
 def test_frame_write_read_verify(tempdir, input_symbols, input_days,
                                  file_scheme,
                                  input_columns, partitions, filters):
+    if os.name == 'nt':
+        pytest.xfail("Partitioning folder names contain special characters which are not supported on Windows")
+
     # Generate Temp Director for parquet Files
     fdir = str(tempdir)
     fname = os.path.join(fdir, 'test')

--- a/fastparquet/test/test_read.py
+++ b/fastparquet/test/test_read.py
@@ -319,7 +319,7 @@ def test_multi_index(tempdir):
     assert dg.index.levels[0].name == 'a'
     assert dg.index.levels[0].dtype == '<M8[ns]'
     assert dg.index.levels[1].name == 'b'
-    assert dg.index.levels[1].dtype == int
+    assert dg.index.levels[1].dtype == np.int64
 
 
 def test_multi_index_category(tempdir):

